src/transactions/test/InvokeHostFunctionTests.cpp

```diff
@@ -388,6 +389,118 @@ TEST_CASE("Native stellar asset contract",
+        {
+            INFO("transfer from muxed account to muxed account");
+            REQUIRE(client.transfer(
+                a1, makeMuxedAccountAddress(a2.getPublicKey(), 123'456'789),
+                100'000'000, std::numeric_limits<uint64_t>::max()));
+            REQUIRE(*client.lastEvent() ==
+                    client.makeTransferEvent(
+                        a1Address, a2Address, 100'000'000,
+                        std::numeric_limits<uint64_t>::max(), 123'456'789));
+        }
+        {
+            INFO("transfer from muxed account to contract");
+            REQUIRE(client.transfer(a2, transferContract.getAddress(),
+                                    200'000'000, 0));
+            REQUIRE(*client.lastEvent() ==
+                    client.makeTransferEvent(a2Address,
+                                             transferContract.getAddress(),
+                                             200'000'000, 0));
+        }
+        {
+            INFO("transfer from contract not supporting muxed accounts")
+            auto contractToAccountSpec =
+                client.defaultSpec()
+                    .setReadOnlyFootprint(client.getContract().getKeys())
+                    .extendReadWriteFootprint(
+                        {client.makeBalanceKey(transferContract.getAddress()),
+                         client.makeBalanceKey(a2Address)});
+            // Muxed destination won't work.
+            REQUIRE(
+                !transferContract
+                     .prepareInvocation(
+                         "transfer_1",
+                         {makeAddressSCVal(client.getContract().getAddress()),
+                          makeAddressSCVal(
+                              makeMuxedAccountAddress(a2.getPublicKey(), 111))},
+                         contractToAccountSpec)
+                     .invoke());
+            // Non-muxed destination will work.
+            REQUIRE(
+                transferContract
+                    .prepareInvocation(
+                        "transfer_1",
+                        {makeAddressSCVal(client.getContract().getAddress()),
+                         makeAddressSCVal(a2Address)},
+                        contractToAccountSpec)
+                    .invoke());
+        }
+        {
+            INFO("transfer from account to muxed account");
+            REQUIRE(client.transfer(
+                a1,
+                makeMuxedAccountAddress(a2.getPublicKey(),
+                                        123'456'789'123'456'789ULL),
+                300'000'000));
+            REQUIRE(*client.lastEvent() ==
+                    client.makeTransferEvent(a1Address, a2Address, 300'000'000,
+                                             std::nullopt,
+                                             123'456'789'123'456'789ULL));
+        }
+        {
+            INFO("transfer to liquidity pool fails");
+            REQUIRE(!client.transfer(a1, makeLiqudityPoolAddress(PoolID()), 1));
```

Comment:
It it worth testing the same but also with `from` instead of `to`?

Reply:
That would be too cumbersome to setup with our test framework and also it wouldn't add much coverage. I think this should be sufficient.